core/src/main/java/org/apache/carbondata/core/datastore/compression/AbstractCompressor.java

```diff
@@ -48,10 +49,11 @@
   }
 
   @Override
-  public byte[] compressInt(int[] unCompInput) {
-    ByteBuffer unCompBuffer = ByteBuffer.allocate(unCompInput.length * ByteUtil.SIZEOF_INT);
+  public ByteBuffer compressInt(int[] unCompInput) {
+    ByteBuffer unCompBuffer = ByteBuffer.allocateDirect(unCompInput.length * ByteUtil.SIZEOF_INT);
```

Comment:
`DirecByteBuffer` are tricky, they reside outside JVM. we need to clean by reflection. Hence it is not used in our code. (can use unsafe if we still need to offheap memory)
DirecByteBuffers are useful only while interacting with native libraries. 
So, I suggest we can use `ByteBuffer` instead of `DirectByteBuffer`

https://stackoverflow.com/questions/5670862/bytebuffer-allocate-vs-bytebuffer-allocatedirect

@ravipesala , @kumarvishal09 what's your opinion on this ?

Reply:
DirectByteBuffer will be GC collected if there are no object reference to it. In our compression case, this object is temporary. So I think it is ok. @ajantha-bhat